### ISSUE TITLE
Instrument the Django REST Framework (DRF)

### ISSUE THREAD

**Issue by itsderek23** [scoutapp/scout_apm_python#102]
Provide more detailed instrumentation of the Django REST Framework (DRF).

http://www.django-rest-framework.org/

**Comment by adamchainz** [comment]
@cschneid wrote on Slack that one customer reported no controller instrumentation for DRF - I'll investigate!

**Comment by adamchainz** [comment]
I set up a DRF test app ( https://github.com/adamchainz/scout-test-apps/tree/master/django_drf ) and found the current Django integration will track DRF `APIView`s and `ViewSet`s like other Django views. Version 2.2.0 that was just released includes some fixes from #203 that help here.  During this research I found two more things we could track:

* Setting the controller method to include which method inside a `ViewSet` was called
* Hooking into DRF's internal exception handling ( https://github.com/encode/django-rest-framework/blob/3.10.1/rest_framework/views.py#L436 ) so these responses are also tracked as errors

I'm sure there are other places in DRF we could add extra instrumentation, any comments appreciated.
